quadkey/library/js/quadint.js

```diff
@@ -0,0 +1,190 @@
+};
+module.exports.ZXYFromQuadint = this.ZXYFromQuadint;
+
+/**
+ * get quadint for location at specific zoom level
+ * @param  {geocoord} location  location coordinates to convert to quadint
+ * @param  {number}   zoom      map zoom level of quadint to return
+ * @return {string}             quadint the input location resides in for the input zoom level
+ */
+quadintFromLocation = function(long, lat, zoom) {
+    if (zoom < 0 || zoom > 29) {
+        throw new Error('Wrong zoom');
+    }
+    const tile = tilebelt.pointToTile(long, lat, zoom - 1);
+    return quadintFromZXY(zoom, tile[0], tile[1]);
+};
+module.exports.quadintFromLocation = this.quadintFromLocation;
+
+/**
+ * convert quadkey into a quadint
+ * @param  {string} quadkey     quadkey to be converted
+ * @return {int}                quadint
+ */
+quadintFromQuadkey = function(quadkey) {
+    const z = quadkey.length;
+    const tile = tilebelt.quadkeyToTile(quadkey);
+    return quadintFromZXY(z, tile[0], tile[1]);
+};
+module.exports.quadintFromQuadkey = this.quadintFromQuadkey;
+
+/**
+ * convert quadint into a quadkey
+ * @param  {int} quadint    quadint to be converted
+ * @return {string}         quadkey
+ */
+quadkeyFromQuadint = function(quadint) {
+    const tile = ZXYFromQuadint(quadint);
+    return tilebelt.tileToQuadkey([tile.x, tile.y, tile.z]);
+};
+module.exports.quadkeyFromQuadint = this.quadkeyFromQuadint;
+
+/**
+ * get the bounding box for a quadint in location coordinates
+ * @param  {int} quadint    quadint to get bounding box from
+ * @return {bbox}           bounding box for the input quadint
+ */
+bbox = function(quadint) {
+    const tile = ZXYFromQuadint(quadint);
+    return tilebelt.tileToBBOX([tile.x, tile.y, tile.z]);
+};
+module.exports.bbox = this.bbox;
+
+/**
+ * get the GeoJSON with the bounding box for a quadint in location coordinates
+ * @param  {int} quadint    quadint to get bounding box from
+ * @return {GeoJSON}        GeoJSON with the bounding box for the input quadint
+ */
+quadintToGeoJSON = function(quadint) {
+    const tile = ZXYFromQuadint(quadint);
+    return tilebelt.tileToGeoJSON([tile.x, tile.y, tile.z]);
+};
+module.exports.quadintToGeoJSON = this.quadintToGeoJSON;
+
+/**
+ * returns the sibling of the given quadint and will wrap
+ * @param  {int} quadint      key to get sibling of
+ * @param  {string} direction direction of sibling from key
+ * @return {int}              sibling key
+ */
+sibling = function(quadint, direction) {
+    const tile = ZXYFromQuadint(quadint);
+    const z = tile.z;
+    let x = tile.x;
+    let y = tile.y;
+    const tilesPerLevel = 2 << (z - 1);
+    if (direction === 'left') {
+        x = x > 0 ? x - 1 : tilesPerLevel - 1;
+    }
+    if (direction === 'right') {
+        x = x < tilesPerLevel - 1 ? x + 1 : 0;
+    }
+    if (direction === 'up') {
+        y = y > 0 ? y - 1 : tilesPerLevel - 1;
+    }
+    if (direction === 'down') {
+        y = y < tilesPerLevel - 1 ? y + 1 : 0;
+    }
```

Comment:
I think checking the direction is one of the valid values here would be best, otherwise you get the same input and no error (e.g. "Left" won't work as expected).

Reply:
oki doki. I'm going to check for expected values. Do you like the idea of doing also a toLowerCase? or should be less flexible and not allow capital letters?

Reply:
Doing `toLowerCase` seems llike a good idea